src/kernel/src/arch/aarch64/context.rs

```diff
@@ -71,25 +78,34 @@ impl ArchContext {
         Self::new_kernel()
     }
 
-    #[allow(named_asm_labels)]
     pub fn switch_to(&self) {
+        unsafe {
+            Self::switch_to_target(&self.target);
+        }
+    }
+
+    #[allow(named_asm_labels)]
+    /// Switch to a target context.
+    ///
+    /// # Safety
+    /// This function must be called with a target that comes from an ArchContext that lives long enough.
+    pub unsafe fn switch_to_target(tgt: &ArchContextTarget) {
+        // TODO: handle requests to switch to current page tables by just ignoring.
```

Comment:
What do we mean by "current" page tables? ARM splits up the the address space between kernel (high memory) and user (low memory). Should the switch we care about in this function be the view from the kernel, user, or both?

Reply:
It should only switch the user tables.